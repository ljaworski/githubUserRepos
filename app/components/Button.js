import React from "react";
import PropTypes from "prop-types";
import { Button as NativeBaseButton, Text } from "native-base";

export default function Button({ label, onPress, disabled = false, ...props }) {
  return (
    <NativeBaseButton block onPress={onPress} disabled={disabled} {...props}>
      <Text>{label}</Text>
    </NativeBaseButton>
  );
}

Button.propTypes = {
  label: PropTypes.string,
  onClick: PropTypes.func,
  disabled: PropTypes.bool
};
