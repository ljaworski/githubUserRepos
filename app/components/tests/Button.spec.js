import React from "react";
import Button from "../Button";
import renderer from "react-test-renderer";

const props = {
  label: "Test",
  onClick: () => {}
};

test("renders correctly", () => {
  const button = renderer.create(<Button {...props} />).toJSON();
  expect(button).toMatchSnapshot();
});
