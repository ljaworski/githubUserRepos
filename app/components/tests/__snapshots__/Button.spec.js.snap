// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#007aff",
      "borderBottomWidth": null,
      "borderColor": "#007aff",
      "borderLeftWidth": null,
      "borderRadius": 5,
      "borderRightWidth": null,
      "borderTopWidth": null,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 45,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#fff",
        "fontFamily": "System",
        "fontSize": 16.5,
        "marginLeft": 0,
        "marginRight": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    uppercase={false}
  >
    Test
  </Text>
</View>
`;
